src/main/java/seedu/address/logic/commands/AddLogCommand.java

```diff
@@ -52,21 +58,58 @@
     public AddLogCommand(Index index, AddLogDescriptor addLogDescriptor) {
         requireAllNonNull(index, addLogDescriptor);
         this.index = index;
+        this.personWithNameToAddLog = null;
         this.addLogDescriptor = addLogDescriptor;
+        this.byName = false;
     }
 
+    /**
+     * Creates an AddLogCommand to add the specified {@code Log} to the specified
+     * {@code Person}.
+     */
+    public AddLogCommand(Name name, AddLogDescriptor addLogDescriptor) {
+        requireAllNonNull(name, addLogDescriptor);
+        this.personWithNameToAddLog = new Person(name);
+        this.index = null;
+        this.addLogDescriptor = addLogDescriptor;
+        this.byName = true;
+    }
+
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        // get list of persons from model
-        List<Person> lastShownList = model.getFilteredPersonList();
+        Person personToEdit;
+
+        if (this.byName) {
+
+            // find person with same name
+            List<Person> personsToEdit = model.getAddressBook()
```

Comment:
Would it be better to delegate this logic to a method in the model?

Reply:
Yes, I think so, but at the moment there's room for refactoring this part out across multiple methods - we can do that in the next version at one shot, what do you think?

Reply:
Yep sure, just make a note somewhere